lib/grape-swagger.rb

```diff
@@ -50,6 +50,22 @@ def combine_namespaces(app)
         end
       end
 
+      def get_non_nested_params(params)
+        params.each do |param, value|
+          is_nested_param = /^#{ Regexp.quote param }\[.+\]$/
+          params.each do |p, _|
+            if p.match(is_nested_param)
+              params[p][:required] = false unless value[:required].presence
+            end
+          end
+        end
+
+        params.reject do |param, _|
+          is_nested_param = /^#{ Regexp.quote param }\[.+\]$/
+          params.keys.any? { |p| p.match is_nested_param }
+        end
```

Comment:
You should be able to combine `each` and `reject`, there's a bunch of duplicate processing here, notably `is_nested_param`.